cmd/cli/docker/docker_run.go

```diff
@@ -139,101 +134,127 @@ func dockerRun(cmd *cobra.Command, cmdArgs []string, opts *DockerRunOptions) err
+func CreateJobDocker(image string, parameters []string, opts *DockerRunOptions) (*models.Job, error) {
+	engineSpec, err := models.DockerSpecBuilder(image).
+		WithParameters(parameters...).
+		WithWorkingDirectory(opts.WorkingDirectory).
+		WithEntrypoint(opts.Entrypoint...).
+		WithEnvironmentVariables(opts.SpecSettings.EnvVar...).Build()
 	if err != nil {
 		return nil, err
 	}
 
-	labels, err := parse.Labels(ctx, opts.SpecSettings.Labels)
-	if err != nil {
-		return nil, err
+	// TODO(forrest) [refactor]: this logic is duplicated in wasm_run
+	resultPaths := make([]*models.ResultPath, 0, len(opts.SpecSettings.OutputVolumes))
+	for name, path := range opts.SpecSettings.OutputVolumes {
+		resultPaths = append(resultPaths, &models.ResultPath{
+			Name: name,
+			Path: path,
+		})
 	}
 
-	spec, err := legacy_job.MakeDockerSpec(
-		image, opts.WorkingDirectory, opts.Entrypoint, opts.SpecSettings.EnvVar, parameters,
-		legacy_job.WithResources(
-			opts.ResourceSettings.CPU,
-			opts.ResourceSettings.Memory,
-			opts.ResourceSettings.Disk,
-			opts.ResourceSettings.GPU,
-		),
-		legacy_job.WithNetwork(
-			opts.NetworkingSettings.Network,
-			opts.NetworkingSettings.Domains,
-		),
-		legacy_job.WithTimeout(opts.SpecSettings.Timeout),
-		legacy_job.WithInputs(opts.SpecSettings.Inputs.Values()...),
-		legacy_job.WithOutputs(outputs...),
-		legacy_job.WithAnnotations(labels...),
-		legacy_job.WithNodeSelector(nodeSelectorRequirements),
-		legacy_job.WithDeal(
-			opts.DealSettings.TargetingMode,
-			opts.DealSettings.Concurrency,
-		),
-	)
+	task, err := models.NewTaskBuilder().
+		Name("TODO").
+		Engine(engineSpec).
+		Publisher(opts.SpecSettings.Publisher.Value()).
+		ResourcesConfig(&models.ResourcesConfig{
+			CPU:    opts.ResourceSettings.CPU,
+			Memory: opts.ResourceSettings.Memory,
+			Disk:   opts.ResourceSettings.Disk,
+			GPU:    opts.ResourceSettings.GPU,
+		}).
+		InputSources(opts.SpecSettings.Inputs.Values()...).
+		ResultPaths(resultPaths...).
+		Network(&models.NetworkConfig{
+			Type:    opts.NetworkingSettings.Network,
+			Domains: opts.NetworkingSettings.Domains,
+		}).
+		Timeouts(&models.TimeoutConfig{ExecutionTimeout: opts.SpecSettings.Timeout}).
+		Build()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create job: %w", err)
+	}
 
-	// Publisher is optional and we won't provide it if not specified
-	p := opts.SpecSettings.Publisher.Value()
-	if p != nil {
-		spec.Publisher = p.Type //nolint:staticcheck
-		spec.PublisherSpec = *p
+	labels, err := parse.StringSliceToMap(opts.SpecSettings.Labels)
+	if err != nil {
+		return nil, fmt.Errorf("parseing job labels: %w", err)
 	}
 
+	constraints, err := parse.NodeSelector(opts.SpecSettings.Selector)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("parseing job contstrints: %w", err)
+	}
+	job := &models.Job{
+		Name:        "TODO",
+		Namespace:   "TODO",
+		Type:        models.JobTypeBatch,
+		Priority:    0,
```

Comment:
I believe it is assumed that docker and wasm jobs are batch when run over the CLI, but we could easily change that. Additionally priority doesn't have a flag. Gut says we need flags for these - ideas on good default values appreciated.

Reply:
there is a `--target` flag that controls this:
```
      --target all|any                          Whether to target the minimum number of matching nodes ("any") (default) or all matching nodes ("all") (default any)
```

any is a batch job, where all is an ops job

Reply:
My comment here was with respect to the priority fields. Are we okay adding a flag called `--priority` that sets their field?

With respect to the job type, currently controlled via `--target`:
Would we be willing to drop the target flag in favor of a `--type` flag whose default value is `batch` and other possible values are `service`, `ops`, and `daemon`? This would bring the CLI more inline with the fields of the job spec.

Reply:
lets do the migration one step at a time. Lets avoid exposing or renaming flags as this PR's main focus should be internal and to just switch to new APIs.

To answer your questions though, `--priority` is not fully implemented as the compute nodes for example don't respect priority when queueing executions locally in their buffer. Will make sense to expose priority, blog about it and make sure it is documented when it is fully implemented. We should use whatever default value `bacalhau job run` has.

As for `--target`, we don't and should not support service or daemon jobs using imperative job submissions (i.e. `docker run` and `wasm run`). We should allow users to easily update their long running jobs, compare what is deployed in the network and what their local job spec says, and utilize gitops to deploy and update jobs based on committed specs. All that is not feasible with imperative job submissions, and we shouldn't make it easy for users to do bad things to their network.